torch/distributed/distributed_c10d.py

```diff
@@ -3340,6 +3340,8 @@ def gather(tensor, gather_list=None, dst=0, group=None, async_op=False):
     """
     Gathers a list of tensors in a single process.
 
+    This method requires all tensors to be the same size on each process.
```

Comment:
Nit: update the docstring for `gather_list` below too.  Maybe its worth also keeping a pointer up at the top since this has been confusing, but it should normally be described in the Args section.